frontend/javascripts/admin/task/task_create_form_view.js

```diff
@@ -274,12 +276,30 @@ class TaskCreateFormView extends React.PureComponent<Props, State> {
                           }),
                         ));
 
-                      if (response != null && response.dataSetName != null) {
-                        this.props.form.setFieldsValue({ dataSet: response.dataSetName });
+                      if (annotationResponse != null && annotationResponse.dataSetName != null) {
+                        this.props.form.setFieldsValue({
+                          dataSet: annotationResponse.dataSetName,
+                        });
                         return callback();
                       } else {
-                        this.props.form.setFieldsValue({ dataSet: null });
-                        return callback("Invalid base annotation id.");
+                        const taskResponse = await tryToAwaitPromise(
+                          getTask(value, {
+                            showErrorToast: false,
+                          }),
+                        );
+                        if (
+                          taskResponse != null &&
+                          taskResponse.dataSet != null &&
+                          _.isEqual(taskResponse.status, { open: 0, active: 0, finished: 1 })
+                        ) {
+                          this.props.form.setFieldsValue({
+                            dataSet: taskResponse.dataSet,
+                          });
+                          return callback();
```

Comment:
My personal preference is to not use else-branches if the corresponding if-branch ends with an `return`. That way, you can hoist lines 285 to 302 out of the else branch as well as lines 300 to 302. This effectively unflattens the code and makes it more readable (at least, in my opinion :)).